### ISSUE TITLE
[question] VecFrameStack with recurrent policies?

### ISSUE THREAD

**Issue by andytwigg** [openai/baselines#663]
Hi,
When training atari envs with ppo2 and a recurrent policy, it looks like we still use `VecFrameStack(4)`. This way, it seems like the last 4 frames are consumed by each step of the RNN.  Is there a reason for this, or have I missed something?

**Comment by pzhokhov** [comment]
That is correct, even with recurrent policies we still use stacks of frames. On a handwavy level, processing stacks of four frames through CNN should capture fast dynamics (something that happens within these 4 frames), whereas recurrent part of the network can handle longer time horizons @joschu, @olegklimov or @unixpickle may provide a more principled answer. That being said, I'd be curious to see if removing framestacking hurts performance with recurrent policies.

**Comment by aznshodan** [comment]
Is there a reason to why you guys use VecFrameStack instead of wrap_deepmind(frame_stack=True)? Shouldn't wrap_deepmind give out the same observation as VecFrameStack when frame_stack is set to True?

**Comment by unixpickle** [comment]
@aznshodan you are correct that the two ways of doing frame stack are semantically equivalent. They do exactly the same thing. The difference is performance: by using `VecFrameStack`, less data has to be sent through IPC (inter-process communication) from the environment sub-processes to the main process.
